modules/geo-layers/src/tile-layer/tileset-2d.js

```diff
@@ -170,6 +175,23 @@ export default class Tileset2D {
 
       // isSelected used in request scheduler
       tile.isSelected = tile.state === TILE_STATE_SELECTED;
+
+      // Keep track of all the ongoing requests
+      if (tile.isLoading) {
+        ongoingRequestCount++;
+        if (!tile.isSelected) {
+          abortCandidates.push(tile);
+        }
+      }
+    }
+
+    if (maxRequests > 0 && ongoingRequestCount > maxRequests) {
+      // There are too many ongoing requests, see if any can be aborted
+      const tilesToAbort = abortCandidates.slice(0, ongoingRequestCount - maxRequests);
```

Comment:
I don't understand why this would ever be nonzero. I thought the idea of the `RequestScheduler` was to release a max of `maxRequests` at a time from the queue. So if `maxRequests` is 6, there should never be more than 6 tiles at one time where `isLoading` is `true`.

In my mind the preferred implementation is just:
```js
if (tile.isLoading && !tile.isSelected) {
  tile.abort();
}
```

Reply:
In this case, assuming that the user's `getTileData` implements `signal` checking, then `getTileData` would return immediately and the `finally` block would call `requestToken.done();`, which would then allow the `RequestScheduler` to release a new tile request.

Reply:
There could be requests loading but not selected (e.g. user panned a bit and panned back), in this case we wouldn't want to abort them unless we needed to free up some requests.

Reply:
> user panned a bit and panned back

That's a valid scenario that I also had in mind. I think instead of using `ongoingRequestCount`, you should use [`requestScheduler.requestQueue.length`](https://github.com/visgl/loaders.gl/blob/62e52a6e914f517f1f1da967b9a9bd81f9ea93d5/modules/loader-utils/src/lib/request-utils/request-scheduler.js#L26), which will give the total number of requests that are queued, instead of just the number that have already been started.

Reply:
is `requestScheduler.requestQueue.length` the include requests that have started but aren't yet done or just the number that are queued (not started)?

Reply:
Just abort all tiles out of view if you there's a queue?

```js
const nTilesQueued = this.requestScheduler.requestQueue.length;
if (maxRequests > 0 && nTilesQueued > maxRequests) {
  this._cache
  .values()
  .filter(tile => tile.isLoading && !tile.isSelected)
  .map(tile => tile.abort);
}
```

Reply:
We could do that, but we should only abort the number we need to get back under the max. For example, no need to abort all 6 tiles out of view just because one new one came into view and is queue.

Reply:
The thing with this approach is that we have to iterate the cache twice. With the code I have in the PR right now, we only iterate the cache once to build up the tiles to abort.

Reply:
Without having to create new arrays:

```js
if (maxRequests > 0) {
  while (ongoingRequestCount > maxRequests && abortCandidates.length) {
    const tile = abortCandidates.shift();
    tile.abort();
    ongoingRequestCount--;
  }
}
```

Reply:
Made the update. Hopefully this'll work.